molecule/provisioners.py

```diff
@@ -274,13 +266,12 @@ def up(self, no_provision=True):
         self._vagrant.up(no_provision)
 
     def destroy(self):
-        self._vagrant.halt()
-        self._vagrant.destroy()
-        os.remove(self.m._config.config['molecule']['vagrantfile_file'])
-
-    def halt(self):
         self._write_vagrant_file()
-        self._vagrant.halt()
+        for status in self._vagrant.status():
```

Comment:
Would be nice if we can look at the state file's status vs another heavy call to `vagrant status`.